### ISSUE TITLE
browserify?

### ISSUE THREAD

**Issue by pkosenko** [futurepress/epub.js#312]
I see that you are using the RSVP library to help load ebooks, since you want to use promises for the large latency that would be involved.

But I am not sure why the RSVP library is included whole hog at the top of the epub.js file rather than as an include (maybe using browserify?).  I'm not an expert, but I recently attended a talk about it.  It is a bundler program.

http://browserify.org/

https://github.com/substack/browserify-handbook

**Comment by fchasen** [comment]
Currently, we are only targeting browsers so no plan to switch from bundling in the grunt task yet but definitely something to consider for future versions of the library that work in node as well as in the browser.

Additionally we use some of the bundled events library in RSVP extensively as well as some of the less standard promise features of RSVP such as creating deferred promises, so you couldn't just use a different promise library.

**Comment by davidguttman** [comment]
To clear things up, `browserify` (and `webpack`) allow you to write your code using CommonJS `require` or ES `import` so that you don't have to rely on globals declared in a specific order. For example, instead a person having to list `RSVP` ahead of `reader.js` because `reader.js` needs `RSVP`, `reader.js` could `require/import` `rsvp` and `browserify/webpack` could see that it was a dependency and would be able to bundle it automatically (without a human having to worry about concatenation order or config files).

The connection to node.js is simply that a lot of modules are *just JavaScript* so they can work in either environment, and `browserify` is happy to oblige.

Anyways, I saw this module on npm and figured I could just use it via `browserify`, but I got the following errors:

```
Error: module "connect" not found from "/Users/..."
Error: module "optimist" not found from "/Users/..."
Error: module "portfinder" not found from "/Users/..."
```

I then tried to just `require('epub.js/build/epub.min.js')` but then I get `Uncaught ReferenceError: RSVP is not defined`
